docs/api/python/rnn.md

```diff
@@ -0,0 +1,339 @@
+# RNN Cell API
+
+```eval_rst
+.. currentmodule:: mxnet.rnn
+```
+
+## Overview
+
+The `rnn` module includes the recurrent neural network (RNN) cell APIs, a suite of tools for building an RNN's symbolic graph.
+```eval_rst
+.. note:: The `rnn` module offers higher-level interface while `symbol.RNN` is a lower-level interface. The cell APIs in `rnn` module are easier to use in most cases.
+```
+
+## The `rnn` module
+
+### Cell interfaces
+
+```eval_rst
+.. autosummary::
+    :nosignatures:
+
+    BaseRNNCell.__call__
+    BaseRNNCell.unroll
+    BaseRNNCell.reset
+    BaseRNNCell.begin_state
+    BaseRNNCell.unpack_weights
+    BaseRNNCell.pack_weights
+```
+
+When working with the cell API, the precise input and output symbols
+depend on the type of RNN you are using. Take Long Short-Term Memory (LSTM) for example:
+
+```python
+import mxnet as mx
+# Shape of 'step_data' is (batch_size,).
+step_input = mx.symbol.Variable('step_data')
+
+# First we embed our raw input data to be used as LSTM's input.
+embedded_step = mx.symbol.Embedding(data=step_input, \
+                                    input_dim=input_dim, \
+                                    output_dim=embed_dim)
+
+# Then we create an LSTM cell.
+lstm_cell = mx.rnn.LSTMCell(num_hidden=50)
+# Initialize its hidden and memory states.
+# 'begin_state' method takes an initialization function, and uses 'zeros' by default.
+begin_state = lstm_cell.begin_state()
+```
+
+The LSTM cell and other non-fused RNN cells are callable. Calling the cell updates it's state once. This transformation depends on both the current input and the previous states. See this [blog post](http://colah.github.io/posts/2015-08-Understanding-LSTMs/) for a great introduction to LSTM and other RNN.
+```python
+# Call the cell to get the output of one time step for a batch.
+output, states = lstm_cell(embedded_step, begin_state)
+
+# 'output' is lstm_t0_out_output of shape (batch_size, hidden_dim).
+
+# 'states' has the recurrent states that will be carried over to the next step,
+# which includes both the "hidden state" and the "cell state":
+# 'lstm_t0_out_output' has shape (batch_size, hidden_dim), and 'lstm_t0_state_output' has shape (batch_size, hidden_dim).
+```
+
+Most of the time our goal is to process a sequence of many steps. For this, we need to unroll the LSTM according to the sequence length.
+```python
+# Embed a sequence. 'seq_data' has the shape of (sequence_length, batch_size).
```

Comment:
This section is just about embedding a variable which doesn't depend on the data iterator. Changing the layout of the data iterator seems to be a topic that should be covered in data I/O APIs page.